src/Raven.Server/Indexing/TempFileCache.cs

```diff
@@ -261,5 +263,10 @@ public void ResetLength()
             _length = 0;
             InnerStream.Position = 0;
         }
+
+        public Stream CreateReaderStream()
+        {
+            return SafeFileStream.Create(InnerStream.SafeFileHandle, FileAccess.Read);
```

Comment:
If I create two files here from the same location, do they share the same underlying file pos pointer, or are they separate?

Reply:
I hope I get it right, but IMO each `SafeFileStream` we create has a separate read position

Reply:
Please do a test to verify that.

Reply:
Done